src/appimagetool_fetch_runtime.cpp

```diff
@@ -281,15 +281,15 @@ bool fetch_runtime(char *arch, size_t *size, char **buffer, bool verbose) {
 
         auto runtimeData = response.data();
 
-        if (runtimeData.size() != response.contentLength()) {
+        if (runtimeData.size() != static_cast<std::vector<char>::size_type>(response.contentLength())) {
```

Comment:
`static_cast<size_t>` would be more readable and match the change on line 290

However curl gives a negative result if the size is not known, which needs to be handled.